pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -2124,22 +2135,25 @@ public void updateLogicalTableConfig(LogicalTableConfig logicalTableConfig)
       logicalTableConfig.setBrokerTenant("DefaultTenant");
     }
 
-    LogicalTableUtils.validateLogicalTableName(
+    LogicalTableConfigUtils.validateLogicalTableConfig(
         logicalTableConfig,
         PinotHelixPropertyStoreZnRecordProvider.forTable(_propertyStore)::exist,
         getAllBrokerTenantNames()::contains,
         _propertyStore
     );
 
-    if (!ZKMetadataProvider.isLogicalTableExists(_propertyStore, tableName)) {
+    LogicalTableConfig oldLogicalTableConfig = ZKMetadataProvider.getLogicalTableConfig(_propertyStore, tableName);
+    if (oldLogicalTableConfig == null) {
       throw new TableNotFoundException("Logical table: " + tableName + " does not exist");
     }
 
     LOGGER.info("Updating logical table {}: Updating logical table config in the property store", tableName);
     ZKMetadataProvider.setLogicalTableConfig(_propertyStore, logicalTableConfig);
 
-    LOGGER.info("Updating logical table {}: Updating BrokerResource for table", tableName);
-    updateBrokerResourceForLogicalTable(logicalTableConfig, tableName);
+    if (oldLogicalTableConfig.getBrokerTenant().equals(logicalTableConfig.getBrokerTenant())) {
```

Comment:
Is this one the opposite of what it should be ? `!equals` instead of `equals` ? If I am reading this right, the broker resource is updated if the tenants are equal.